sdk/lib/cmlib/hiveinit.c

```diff
@@ -16,12 +16,13 @@
  */
 BOOLEAN CMAPI
 HvpVerifyHiveHeader(
-    IN PHBASE_BLOCK BaseBlock)
+    IN PHBASE_BLOCK BaseBlock,
+    IN ULONG HfileType)
 {
     if (BaseBlock->Signature != HV_HBLOCK_SIGNATURE ||
         BaseBlock->Major != HSYS_MAJOR ||
         BaseBlock->Minor < HSYS_MINOR ||
-        BaseBlock->Type != HFILE_TYPE_PRIMARY ||
+        BaseBlock->Type != HfileType ||
```

Comment:
Since you generalized this function to be also usable for `HFILE_TYPE_LOG` types :+1: :
have you checked whether Windows registry LOG files' base block have all the Major and Minor fields set?

Reply:
Base block in log file is copy of original registry base block.

Reply:
OK so it's better to check for this.